### PR TITLE
Module 12

### PR DESCRIPTION
**Pull request by beccar97**
Site can be found at: https://beccar-prod-webapp.azurewebsites.net

### CODE REVIEW

**Review by hugh-emerson — APPROVED** [review]
Looks good. You're leaking a secret but it's a simple fix. After you fix that merge when you're ready. 👍

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    app_command_line = ""
+    linux_fx_version = "DOCKER|beckycarter/todo-app:latest"
+  }
+  app_settings = {
+    "DOCKER_REGISTRY_SERVER_URL" = "https://index.docker.io"
+    "MONGO_CONNECTION_STRING"    = "mongodb://${azurerm_cosmosdb_account.main.name}:${azurerm_cosmosdb_account.main.primary_key}@${azurerm_cosmosdb_account.main.name}.mongo.cosmos.azure.com:10255/${azurerm_cosmosdb_mongo_database.main.name}?ssl=true&replicaSet=globaldb&retrywrites=false&maxIdleTimeMS=120000"
+    "GITHUB_AUTH_CLIENT_ID" = var.github_auth_client_id
+    "GITHUB_AUTH_CLIENT_SECRET" = var.github_auth_client_secret
+    "FLASK_SECRET_KEY" = random_uuid.uuid.result
+    "OAUTHLIB_INSECURE_TRANSPORT" = 1
```
[the remark this answers is not in the document]
Comment (beccar97) [reply main.tf]: I thought that, and I did initially try not including this variable. However despite having the redirect urls using https in the OAuth app set up, I find that when I remove this I get insecure transport errors. Do you have any ideas as to what I might be missing / if no immediate ideas could we possibly schedule a chat to look at it?

File: outputs.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+output "webapp_url" {
+  value = "https://${azurerm_app_service.main.default_site_hostname}"
+}
+output "webhook_url" {
+  value = "https://${azurerm_app_service.main.site_credential[0].username}:${azurerm_app_service.main.site_credential[0].password}@${azurerm_app_service.main.name}.scm.azurewebsites.net/docker/hook"
+}
```
[the remark this answers is not in the document]
Comment (beccar97) [reply outputs.tf]: oops, thanks!

File: main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (hugh-emerson) [reply main.tf]: Hm, plausibly it's to do with the request being forwarded to the container... You could try logging the `request.full_path` that you're giving to `parse_request_uri_response` to see what's going on

You can actually sidestep the insecure transport issue by not using that method at all... just take the code query param yourself and give it to `prepare_token_request`.
But it's probably better in general to ask the library to validate things.

[the remark this answers is not in the document]
Comment (beccar97) [reply main.tf]: Issue was that the way that the request passes through Azure meant it reached the app as a http request. Found a way to get the original protocol from the headers and reconstruct the uri with the right protocol to resolve it

**Pull request merged**